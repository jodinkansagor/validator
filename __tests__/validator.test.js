const { Validator } = require('../lib/validator');

const nameValidator = new Validator('name', {
    type: String,
    required: true
});
  
const ageValidator = new Validator('age', {
    type: Number,
    required: true
});
  
const weightValidator = new Validator('weight', {
    type: String,
});

const dirtValidator = new Validator('dirt', {
    type: Boolean,
});


const dog = {
    name: 'spot',
    age: '5',
    weight: '20 lbs'
};

const cat = {
    name: 'garfield',
    age: 'garfieldAge',
    weight: '20 lbs',
};

const bunny = {
    age: [2, 3],
    weight: '10 lbs', 
};

const bird = {
    name: 'Sylvester',
    age: 6
};

const dirt = {
    name: 'dirt',
    dirt: 15,
};


describe('validator module', () => {

    //1
    it('will throw error if the object type is required and the field is not there', () => {
        expect(() => nameValidator.validate(bunny)).toThrowErrorMatchingSnapshot();
    });

    //2 
    it('properly tells if an object is required but the wrong type', () => {
        expect(() => ageValidator.validate(bunny)).toThrowErrorMatchingSnapshot();
        expect(() => ageValidator.validate(cat)).toThrowErrorMatchingSnapshot();
    });


    //3
    it('properly tells if a object type is valid and is there if required', () => {
        expect(nameValidator.validate(dog)).toEqual('spot');
        expect(ageValidator.validate(dog)).toEqual(5);
    });

    //4 
    it('properly can tell if the field is not required and it is missing', () => {
        expect(weightValidator.validate(bird)).toBeNull();
    });
 
    //5
    it('will throw an error if the field is not required, it is there and is is the wrong type', () => {
        expect(() => dirtValidator.validate(dirt)).toThrowErrorMatchingSnapshot();
    });

    //6
    it('can properly tell if an object is not required, there and the right type', () => {
        expect(weightValidator.validate(dog)).toEqual('20 lbs');
    });

    



    

});
