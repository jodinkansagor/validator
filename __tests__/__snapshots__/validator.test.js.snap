// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validator module properly tells if an object is required but the wrong type 1`] = `"Cannot cast >>2,3<< to Number"`;

exports[`validator module properly tells if an object is required but the wrong type 2`] = `"Cannot cast >>garfieldAge<< to Number"`;

exports[`validator module will throw an error if the field is not required, it is there and is is the wrong type 1`] = `"Cannot cast >>15<< to Boolean"`;

exports[`validator module will throw error if the object type is required and the field is not there 1`] = `"error"`;
